supervisor/snapshots/snapshot.py

```diff
@@ -481,3 +495,28 @@ def restore_repositories(self):
         Return a coroutine.
         """
         return self.sys_store.update_repositories(self.repositories)
+
+    def store_dockerconfig(self):
+        """Store the configuration for Docker."""
+        self.docker = {
+            ATTR_REGISTRIES: {
+                registry: {
+                    ATTR_USERNAME: credentials[ATTR_USERNAME],
+                    ATTR_PASSWORD: self._encrypt_data(credentials[ATTR_PASSWORD]),
+                }
+                for registry, credentials in self.sys_docker.config.registries.items()
+            }
+        }
+
+    def restore_dockerconfig(self):
+        """Restore the configuration for Docker."""
+        self.sys_docker.config.registries.update(
+            {
+                registry: {
+                    ATTR_USERNAME: credentials[ATTR_USERNAME],
+                    ATTR_PASSWORD: self._decrypt_data(credentials[ATTR_PASSWORD]),
+                }
+                for registry, credentials in self.docker[ATTR_REGISTRIES].items()
```

Comment:
This will throw a `TypeError` if  `self.docker` is not a dict (`None` if not set)?